peak/peak.py

```diff
@@ -1,2 +1,45 @@
+from collections import OrderedDict
+from bit_vector import BitVector
+
 class Peak:
     pass
+
+
+def name_outputs(**outputs):
+    """Decorator meant to apply to any function to specify output types
+    The output types will be stored in fn.__peak_outputs__
+    The input types will be stored in fn.__peak_inputs__
+    Will verify that all the inputs have type annotations
+    Will also verify that the outputs of running fn will have the correct number of bits
+    """
+    def decorator(call_fn):
+        def call_wrapper(*args,**kwargs):
+            results = call_fn(*args,**kwargs)
+            single_output = not isinstance(results,tuple)
+            if single_output:
+                results = (results,)
+            for i, (oname,otype) in enumerate(outputs.items()):
+                assert results[i].num_bits == otype(0).num_bits, f"{results[i].num_bits} != {otype(0).num_bits}"
+            if single_output:
+                results = results[0]
+            return results
+
+        #Set all the outputs
+        call_wrapper.__peak_outputs__ = OrderedDict()
+        for oname,otype in outputs.items():
+            assert issubclass(otype,BitVector), f"{oname} is not a Bitvector class"
+            call_wrapper.__peak_outputs__[oname] = otype
```

Comment:
`__peak_outputs__` should be renamed.  Use of dunder names is reversed for magic methods and attributes by PEP 8.
https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles